client/src/components/base/Button/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+
+type ButtonTheme = 'normal' | 'github' | 'dark' | 'white';
+type ButtonType = 'button' | 'submit' | 'reset';
+type ButtonSize = 'xlarge' | 'large' | 'medium' | 'small';
+
+type ButtonProps = {
+  size: ButtonSize;
+  value: string;
+  type: ButtonType;
+  onClick: (e: React.MouseEvent) => void;
+  theme: ButtonTheme;
+};
+
+const Button = ({ size, value, type, onClick, theme }: ButtonProps) => {
+  return (
+    <ButtonContainer size={size}>
+      <CustomButton theme={theme} onClick={onClick} type={type}>
+        {theme === 'github' ? <i className="fab fa-github"></i> : ''}
+        {value}
+      </CustomButton>
+    </ButtonContainer>
+  );
+};
+
+export default Button;
+
+type ButtonContainerProps = {
+  size: string;
+};
+
+type CustomButtonProps = {
+  theme: string;
+};
+
+type Color = string;
+const buttonBlack: Color = '#333';
+const buttonDarkBlack: Color = '#000';
+const buttonWhite: Color = '#fff';
+const borderGrey: Color = '#cccccc';
+
+const ButtonContainer = styled.div<ButtonContainerProps>`
+  ${({ size }) => getButtonSize(size)}
+  i {
+    margin-right: 5px;
+  }
+`;
+
+const getButtonSize = (size: string) => {
+  if (size === 'xlarge') {
+    return css`
+      width: 25rem;
+      height: 5rem;
+      button {
+        font-size: 1.5rem;
+      }
+    `;
+  } else if (size === 'large') {
+    return css`
+      width: 25rem;
+      height: 3rem;
+    `;
+  } else if (size === 'medium') {
+    return css`
+      width: 18rem;
+      height: 3rem;
+    `;
+  } else if (size === 'small') {
+    return css`
+      width: 10rem;
+      height: 3rem;
+      button {
+        font-size: 0.8rem;
+      }
+    `;
+  }
+};
+
+const CustomButton = styled.button<CustomButtonProps>`
+  width: 100%;
+  height: 100%;
+  color: #fff;
+  border: none;
+  font-size: 1rem;
+  cursor: pointer;
+  padding: 0 10px;
+  transition: all 0.2s;
+  ${({ theme }) => getButtonTheme(theme)};
+`;
+
+const getButtonTheme = (theme: string) => {
```

Comment:
```suggestion
  ${getButtonTheme};
`;

const getButtonTheme = ({theme}: CustomButtonProps) => {
```

아마 이렇게도 사용하실 수 있을겁니다. 
참고하세용~

Reply:
깔끔하니 좋네요!